app/controllers/posts_controller.rb

```diff
@@ -13,21 +15,47 @@ def index
   end
 
   def update
-    @post = Post.find(params[:id])
     @post.update_attributes(message: post_params)
     redirect_to posts_url
   end
 
   def edit
-    @post = Post.find(params[:id])
+  end
+
+  def destroy
```

Comment:
Maybe this could be done in the view, where by the edit/delete buttons are only visable to those who created the post?